src/Admin/Settings.php

```diff
@@ -99,10 +99,13 @@ public function get_header( $name ) {
 	 */
 	public function plausible_analytics_settings_page() {
 		$settings             = Helpers::get_settings();
-		$domain               = ! empty( $settings['domain_name'] ) ? $settings['domain_name'] : Helpers::get_domain();
-		$custom_domain_prefix = ! empty( $settings['custom_domain_prefix'] ) ? $settings['custom_domain_prefix'] : 'analytics';
-		$self_hosted_domain   = ! empty( $settings['self_hosted_domain'] ) ? $settings['self_hosted_domain'] : 'example.com';
-		$shared_link          = ! empty( $settings['shared_link'] ) ? $settings['shared_link'] : "https://plausible.io/share/{$domain}?auth=XXXXXXXXXXXX";
+		//echo "<pre>";
+		//print_r($settings);
+		//exit("fdfdd");
+		$domain               = ! empty( $settings['domain_name'] ) ? esc_attr($settings['domain_name']) : Helpers::get_domain();
+		$custom_domain_prefix = ! empty( $settings['custom_domain_prefix'] ) ? esc_attr($settings['custom_domain_prefix']) : 'analytics';
+		$self_hosted_domain   = ! empty( $settings['self_hosted_domain'] ) ? esc_attr($settings['self_hosted_domain']) : 'example.com';
+		$shared_link          = ! empty( $settings['shared_link'] ) ? esc_attr($settings['shared_link']) : "https://plausible.io/share/{$domain}?auth=XXXXXXXXXXXX";
```

Comment:
@Ankur0812 This field contains URL. So, use `esc_url` function instead of `esc_attr`

Reply:
@Ankur0812 Also, there is no need to add escaping function here. Do late escaping, in which you need to use escape function when you output these variables

Reply:
I already tried but issue is that we escaping later then display prompt window not working.
if you try <img src="" onprompt=(8)> then escaping on output variable, then prompt window open.

That's why I escaped above.